src/Exception/ExceptionInterface.php

```diff
@@ -8,6 +8,8 @@
  * This interface must be implemented by all exception classes of this library.
  *
  * @author Stefano Arlandini <sarlandini@alice.it>
+ *
+ * @deprecated since version 2.4, to be removed in 3.0
```

Comment:
Why?

Reply:
Because not all our exception classes implement it and I'm not sure it has some value in keeping it. But I'm open to reverting the change if you think it's worth

Reply:
If not all exceptions are implementing it, I think that that's the error that we should fix.

Reply:
The point is, does it makes sense to keep this interface? What are the pros of having it and what are the cons of dropping it?

Reply:
The advantage is giving the possibility to us and to the end user to catch all exceptions that we throw. I don't see downsides in keeping it.

Reply:
Well then, I reverted the change 👍